### PR TITLE
[OpenLR-tool] Mark matched path as golden.

### PR DESCRIPTION
**Pull request by mgsergio**
@ygorshenin

### CODE REVIEW

File: openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp [hunk 1]
```diff
@@ -268,29 +268,32 @@ Qt::ItemFlags TrafficMode::flags(QModelIndex const & index) const
...
-  if (m_currentSegment->HasGoldenPath())
+  if (!m_currentSegment->HasMatchedPath())
   {
-    auto const btn = QMessageBox::question(
-        nullptr,
-        "Override warning",
-        "The selected segment already has a golden path. Do you want to override?");
-    if (btn == QMessageBox::No)
-      return;
+    QMessageBox::information(nullptr, "Error", "The selected segment does not have a matched path");
```
Comment (ygorshenin) [inline openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp]: Comment to the nullptr, please.

File: openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp [hunk 2]
```diff
@@ -502,4 +505,24 @@ void TrafficMode::HandlePoint(m2::PointD clickPoint, Qt::MouseButton const butto
     return;
   }
 }
+
+bool TrafficMode::StartBuildingPathChecks() const
+{
+  CHECK(m_currentSegment, ("A segment should be selected before path building is started."));
+
+  if (m_buildingPath)
```
Comment (ygorshenin) [inline openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp]: Could you please explain why do you mix CHECKs with exceptions? What's the reason to use exceptions here?

File: openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp (hunk 1 above)
Comment (mgsergio) [reply openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp]: Done.

File: openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp (hunk 2 above)
Comment (mgsergio) [reply openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp]: Exception may be handled by ui. The situation that leads to this exception is triggered by a user.
While CHECK(m_currentSegment, (...)) is for catching logical error.

Comment (ygorshenin) [reply openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp]: Okay.

**Pull request merged**